### ISSUE TITLE
EC2 instance type

### ISSUE THREAD

**Issue by mericano1** [npalm/terraform-aws-gitlab-runner#350]
Is there any particular way to pick the runner image / architecture / instance type ? I would like to have some AMD and some ARM64 runners. Is that possible ?

Thanks

**Comment by npalm** [comment]
The runners are just plain ec2 instances managed via docker-machine (legacy). Once you are able to launch via docker-machine ARM instances, it should be no problem.

**Comment by mericano1** [comment]
Hi Nick, thanks for your reply. I found the issue was actually with the helper image. I had to add (terraform file)

```
  helpers = {
    image = "gitlab/gitlab-runner-helper:arm-latest"
  }
```
to the configuration to make it work. Would be good to document that maybe ?

Thanks

**Comment by mericano1** [comment]
Hi @npalm - thanks for getting back to me.
Any idea of where I'd configure those in the terraform files ?

**Comment by kayman-mk** [comment]
Have you checked https://github.com/thejohnfreeman/ami-gitlab-runner ? You could choose the M6g instance type.

**Comment by kayman-mk** [comment]
@mericano1 Can this issue be closed?

**Comment by mericano1** [comment]
Yes, I think so. We eventually found another solution TBH

**Comment by lmilbaum** [comment]
@mericano1 Have you being able to use this module to run ARM based runners? If yes, would you mind share your insights.

**Comment by jessedobbelaere** [comment]
For people that stumble upon this issue. I've switched to ARM64 runners, it was pretty easy. I only had to change the following two:

```hcl
  runner_worker_docker_machine_instance = {
    types     = ["m6g.large"] # Use a graviton EC2 instance
  }

  runner_worker_docker_machine_ami_filter = {
    name = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-arm64-server-*"]
  }
```

I'm building docker images in my Gitlab job for ARM64 architecture, and it's a lot faster not having the QEMU emulation going on 🚀 In the end i've created two runner module instances: one for untagged jobs with the default `amd64` and one with tag `arm64` on the runner config. My gitlab jobs tagged with arm64 can spin up the right runner docker-machine worker now 

```yaml
  tags:
    - arm64
```
